Tinder.DAL/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Tinder.DAL.Interfaces;
+using Tinder.DAL.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Tinder.DAL.Repositories
+{
+    public class GenericRepository<TEntity> : IGenericRepository<TEntity> where TEntity : BaseEntity
+    {
+        protected ApplicationDbContext _context;
+        public GenericRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+        public virtual async Task<TEntity> CreateAsync(TEntity entity, CancellationToken cancellationToken)
+        {
+           await _context.Set<TEntity>().AddAsync(entity, cancellationToken);

```

Comment:
You can move _context.Set<TEntity>() to property, initialize on ctor and re-use it